openapi/spec.go

```diff
@@ -93,7 +93,26 @@ type PathItem struct {
 // other components in the specification, internally and
 // externally.
 type Reference struct {
-	Ref string `json:"$ref" yaml:"$ref"`
+	Ref        string                 `json:"$ref" yaml:"$ref"`
+	Extensions map[string]interface{} `json:"-" yaml:"-"`
```

Comment:
I don't get the rationale for this field addition.
Extensions should only be present on the `Schema` struct. It can then be used inline or be references through the spec components.